### PR TITLE
new: Allow co-branded bcmc

### PR DESCRIPTION
**Pull request by goergisn**
## Summary

<new>

* Bancontact cards are now supported in the `CardComponent`
* Dual-branded Bancontact cards now allow brand selection in the `BCMCComponent`

</new>

- Adding type to BinLookupRequest for correct card brand sorting
- Using CardPaymentMethod's `brands` to configure the `BCMCComponent`

### CODE REVIEW

**Review by descorp — DISMISSED** [review]

File: AdyenCard/Form/FormCardSecurityCodeItem.swift [hunk 1]
```diff
@@ -9,19 +9,41 @@
...
+            switch self {
+            case .required, .optional: return true
+            case .hidden: return false
+            }
+        }
+        
+        internal var isOptional: Bool {
+            switch self {
+            case .optional: return true
+            case .hidden, .required: return false
```
Comment (erenbesel) [inline AdyenCard/Form/FormCardSecurityCodeItem.swift]: Isn't it considered optional if it's hidden as well?

File: AdyenCard/Form/FormCardNumberContainerItem.swift [hunk 2]
```diff
@@ -55,6 +67,12 @@ internal final class FormCardNumberContainerItem: FormItem, AdyenObserver {
     
     internal func update(brands: [CardBrand]) {
         numberItem.update(brands: brands)
+        
+        guard showsSupportedCardLogos else {
```
Comment (erenbesel) [inline AdyenCard/Form/FormCardNumberContainerItem.swift]: exiting here seems dangerous if we want to add some other logic below that could be for both cases. Maybe if else block for this and the below related logic?

File: AdyenCard/Utilities/BinLookup/BinLookupService.swift [hunk 3]
```diff
@@ -27,12 +27,19 @@ internal final class BinLookupService: AnyBinLookupService {
...
         self.publicKey = publicKey
         self.apiClient = apiClient
+        self.binLookupType = binLookupType
     }
     
-    internal func requestCardType(for bin: String, supportedCardTypes: [CardType], caller: @escaping CompletionHandler) {
+    internal func requestCardType(
+        for bin: String,
+        supportedCardTypes: [CardType],
+        caller: @escaping CompletionHandler
```
Comment (erenbesel) [inline AdyenCard/Utilities/BinLookup/BinLookupService.swift]: Not sure why it was called `caller` before but can we rename it to `completion` :D

File: AdyenCard/Form/FormCardSecurityCodeItem.swift (hunk 1 above)
Comment (goergisn) [reply AdyenCard/Form/FormCardSecurityCodeItem.swift]: When re-reading it, it is actually confusing - it was meant to check for the actual enum value `optional` hence the naming. I refactored it to us a switch in place as it's only used once with more supporting context 👌

**Review by erenbesel — DISMISSED** [review]

**Review by erenbesel — APPROVED** [review]

**Review by neelSharma12 — APPROVED** [review]

**Review by descorp — APPROVED** [review]

**Pull request merged**